models/book.go

```diff
@@ -0,0 +1,39 @@
+package models
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+type Book struct {
+	Id             string     `json:"index._id" binding:"required"`
+	Title          string     `json:"title" binding:"required"`
+	AuthorName     string     `json:"author_name"  binding:"required"`
+	Price          float64    `json:"price"  binding:"required"`
+	EbookAvailable bool       `json:"ebook_available"  binding:"required"`
+	PublishDate    CustomTime `json:"publish_date" binding:"required"`
+}
+
+type CustomTime time.Time
+
+const ctLayout = "2006-01-02"
+
+// UnmarshalJSON Parses the json string in the custom format
+func (ct *CustomTime) UnmarshalJSON(b []byte) (err error) {
```

Comment:
Why do you need this? Can't you just use time?

Reply:
The assignment requests use "2006-01-02" format which requires special marshal and unmarshal functions to work